src/components/AnonymousChat/child/ChatReplyPreview.tsx

```diff
@@ -14,29 +18,40 @@ interface ChatReplyPreviewProps {
 }
 
 const ChatReplyPreview = ({type}: ChatReplyPreviewProps) => {
+  const scrollContext = React.useContext(ScrollContext);
   const {replyPreview, clearReplyPreview} = useMessageHook();
+  const isDeleted = replyPreview?.message_type === MESSAGE_TYPE_DELETED;
+
+  const handleTap = () => {
+    if (replyPreview?.id) scrollContext?.handleScrollTo(replyPreview?.id);
+  };
 
   const textContainerStyle = [
-    type === 'SIGNED' ? styles.containerSigned : styles.containerAnon,
+    type === SIGNED ? styles.containerSigned : styles.containerAnon,
     styles.textContainer
   ];
 
+  const messageStyle = [styles.text, isDeleted && styles.deletedText];
+
   return (
     <Animated.View
       entering={SlideInDown.duration(150)}
       exiting={SlideOutDown.duration(80)}
       style={styles.container}>
-      <View style={textContainerStyle}>
+      <TouchableOpacity
+        activeOpacity={replyPreview?.id ? 0.75 : 1}
+        onPress={handleTap}
+        style={textContainerStyle}>
         <View style={styles.chatTitleContainer}>
-          <Text style={styles.userText}>{replyPreview?.username}</Text>
+          <Text style={styles.userText}>{replyPreview?.user?.username}</Text>
           <View style={styles.dot} />
-          <Text style={styles.timeText}>{replyPreview?.time}</Text>
+          <Text style={styles.timeText}>{calculateTime(replyPreview?.updated_at, true)}</Text>
         </View>
 
-        <Text style={styles.text} numberOfLines={1}>
-          {`${replyPreview?.message}`}
+        <Text style={messageStyle} numberOfLines={1}>
+          {replyPreview?.message ?? ''}
         </Text>
-      </View>
+      </TouchableOpacity>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleTap` function scrolls to the message with the id of `replyPreview?.id`. However, there is no check if the message with this id still exists in the chat. If the message was deleted or is not loaded yet (for example, in case of paginated loading), this could lead to unexpected behavior. Consider adding a check if the message with this id exists before scrolling.

```diff
  const handleTap = () => {
    if (replyPreview?.id) {
+     if (messageExists(replyPreview?.id)) {
        scrollContext?.handleScrollTo(replyPreview?.id);
+     } else {
+       // Handle the case when the message does not exist
+     }
    }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->